src/backends/beamcoder.ts

```diff
@@ -242,7 +249,14 @@ export class BeamcoderExtractor extends BaseExtractor implements Extractor {
      * This is the duration of the first video stream in the file expressed in seconds.
      */
     get duration(): number {
-        return this.ptsToTime(this.#demuxer.streams[this.#streamIndex].duration);
+        const stream = this.#demuxer.streams[this.#streamIndex];
+        if (stream.duration !== null) {
+            return this.ptsToTime(stream.duration);
```

Comment:
Hm, when I was playing with libav, I noticed that duration is negative sometimes. are we sure it's not the case here?

Reply:
The duration is "null" in this case.